packages/ag-charts-enterprise/src/features/annotations/annotations.ts

```diff
@@ -183,30 +202,63 @@ export class Annotations extends _ModuleSupport.BaseModuleInstance implements _M
     private readonly colorPicker = new ColorPicker(this.ctx);
     private defaultColor?: string;
 
+    private readonly textInput = new TextInput(this.ctx);
+
     private xAxis?: AnnotationAxis;
     private yAxis?: AnnotationAxis;
 
     constructor(readonly ctx: _ModuleSupport.ModuleContext) {
         super();
+        this.state = this.setupStateMachine();
+        this.setupListeners();
+    }
 
-        this.state = new AnnotationsStateMachine(
-            () => {
-                ctx.cursorManager.updateCursor('annotations');
-                ctx.interactionManager.popState(InteractionState.Annotations);
-                ctx.toolbarManager.toggleGroup('annotations', 'annotationOptions', this.active != null);
-                ctx.tooltipManager.unsuppressTooltip('annotations');
-                for (const annotationType of ANNOTATION_BUTTONS) {
-                    ctx.toolbarManager.toggleButton('annotations', annotationType, { active: false });
-                }
-                this.toggleAnnotationOptionsButtons();
-            },
+    private setupStateMachine() {
+        const { ctx } = this;
+
+        const onEnterIdle = () => {
+            ctx.cursorManager.updateCursor('annotations');
+            ctx.interactionManager.popState(InteractionState.Annotations);
+            ctx.toolbarManager.toggleGroup('annotations', 'annotationOptions', this.active != null);
+            ctx.tooltipManager.unsuppressTooltip('annotations');
+            for (const annotationType of ANNOTATION_BUTTONS) {
+                ctx.toolbarManager.toggleButton('annotations', annotationType, { active: false });
+            }
+            this.toggleAnnotationOptionsButtons();
+        };
+
+        const onExitSingleClick = () => {
+            this.active = this.annotationData.length - 1;
+        };
+
+        return new AnnotationsStateMachine(
```

Comment:
The setup of this state machine is getting a bit unwieldy. I'll look into improving it when working on the editing of text annotations.